Xext/xres.c

```diff
@@ -273,12 +273,28 @@ ProcXResQueryClientResources(ClientPtr client)
 
     FindAllClientResources(resClient, ResFindAllRes, counts);
 
+    xXResType *scratch = calloc(sizeof(xXResType), lastResourceType + 1);
+    if (!scratch) {
+        free(counts);
+        return BadAlloc;
+    }
+
     int num_types = 0;
-    for (int i = 0; i <= lastResourceType; i++) {
-        if (counts[i])
-            num_types++;
+    for (int i = 0; i < num_types; i++) {
```

Comment:
This loop will never execute since num_types is zero?

Reply:
you're right: we need to loop over lastResourceType